site/profile/manifests/core/icinga_master.pp

```diff
@@ -275,6 +275,38 @@
     api_password  => $api_pwd,
     require       => Mysql::Db[$mysql_director_db],
   }
+  ##Director Daemon
+  user { 'icingadirector':
+    ensure => 'present',
+    system => true,
+    shell  => '/bin/false',
+    groups => 'icingaweb2',
+    home   => '/var/lib/icingadirector',
+  }
+  ->file { '/var/lib/icingadirector':
+    ensure => 'directory',
+    mode   => '0750',
+    owner  => 'icingadirector',
+    group  => 'icingaweb2',
+  }
+  class { '::php::globals':
+    php_version => 'rh-php73',
+    rhscl_mode  => 'rhscl',
+  }
+  ->class { '::php':
+    manage_repos => false,
+    extensions   => {
+      'soap'    => {},
+      'process' => {},
+    },
+  }
+  systemd::unit_file { 'icinga-director.service':
+    source  => '/usr/share/icingaweb2/modules/director/contrib/systemd/icinga-director.service',
+    require => User['icingadirector'],
+  }
+  ~> service { 'icinga-director':
```

Comment:
I suspect this can be replaced by `icingaweb2::module::director::service` but, FYI, if the `enable` param is not set on a service resource puppet won't manage that state and its possible the service won't start after a reboot.  The puppet agent should not be relied upon to put the host into a working state, run scripts periodically, etc.

Reply:
as answered before, there's a bug that prevents the correct deployment of the service